packages/xo-server/src/xapi/mixins/pool.mjs

```diff
@@ -55,8 +55,15 @@ const methods = {
 
     await Promise.all(hosts.map(host => host.$call('assert_can_evacuate')))
 
+    // Steps in the RPR : Evacuate hosts, reboot hosts, migrate VMs back, and potentially updateHosts (beforeEvacuateVms and beforeRebootHost)
+    const progressShare = 100 / (3 + !!beforeEvacuateVms + !!beforeRebootHost)
```

Comment:
I don't get this. `!!beforeEvacuateVms` should return a boolean and not a number, no?

What do you think of:
```js
let nSteps = 3
if(beforeEvacuateVms !== undefined) {
  nSteps++
}
...
const progressShare = 100 / nSteps
```

Reply:
It does indeed return a boolean but the addition converts it to 0 or 1.
It was just a way to be more concise, but I don't know if it's a good practice or not.

Reply:
It's not a good practice because it's not easy to understand, especially due to the conversion to boolean and then to number. :slightly_smiling_face:

Maybe adding steps would be enough:

```js
let steps = 3 + Number(beforeEvacuateVms !== undefined) + Number(beforeRebootHost !== undefined)

const progressShare = 100 / steps
```

Reply:
Also, IMHO, `progressShare` is unclear, maybe something like `progressStep` is clearer?